### ISSUE TITLE
Handling Complex Headings

### ISSUE THREAD

**Issue by kylekarpack** [cedricdelpoux/gatsby-source-google-docs#104]
It appears that headings with any sort of complexity (i.e. part bold, part normal) are being split and not fully processed. An example would be a simple document such as this:

![image](https://user-images.githubusercontent.com/2429580/108642702-b4da8d80-745b-11eb-9a86-796631e9ea79.png)

(Available at https://docs.google.com/document/d/1cZ7o50cWtnljBFhLabvvAHRTnSeu13Yr1jHFx0ol5CQ/edit?usp=sharing)

## Expected Behavior
I would expect this to render as:

```markdown
##  **Part Bold** Part Not\n\n\nSome other text\n
```
or in HTML as:
```html
<h2><strong>Part Bold<strong> Part Not</h2>\n<p>Some other text</p>
```

## Actual Behavior
The "Part Bold" item is omitted from the output. Renders as:
```markdown
##  Part Not\n\n\nSome other text\n
```
and
```html
<h2>Part Not</h2>\n<p>Some other text</p>
```

## Info
Here is an example query in Gatsby:
```graphql
{
  googleDocs(slug: {eq: "/debug"}) {
    childMarkdownRemark {
      rawMarkdownBody
      html
    }
    document {
      body {
        content {
          paragraph {
            paragraphStyle {
              namedStyleType
            }
            elements {
              textRun {
                content
              }
            }
          }
        }
      }
    }
  }
}
```
and the resulting output, which shows that the "Part Bold" item is in the Google Docs API response:
```json
{
  "data": {
    "googleDocs": {
      "childMarkdownRemark": {
        "rawMarkdownBody": "##  Part Not\n\n\nSome other text\n",
        "html": "<h2>Part Not</h2>\n<p>Some other text</p>"
      },
      "document": {
        "body": {
          "content": [
            {
              "paragraph": null
            },
            {
              "paragraph": {
                "paragraphStyle": {
                  "namedStyleType": "HEADING_1"
                },
                "elements": [
                  {
                    "textRun": {
                      "content": "Part Bold"
                    }
                  },
                  {
                    "textRun": {
                      "content": " Part Not\n"
                    }
                  }
                ]
              }
            },
            {
              "paragraph": {
                "paragraphStyle": {
                  "namedStyleType": "NORMAL_TEXT"
                },
                "elements": [
                  {
                    "textRun": {
                      "content": "Some other text\n"
                    }
                  }
                ]
              }
            }
          ]
        }
      }
    }
  },
  "extensions": {}
}
```

If you think the expected behavior is correct and can give some pointers as to where to start looking, I'd be happy to submit a pull request with a fix

**Comment by kylekarpack** [comment]
After some experimentation, I found that this has to do with the `demoteHeadings` option. If that option is set to `false`, it appears this issue is mostly resolved. It still doesn't do a great job of processing the internals of a complex heading, but it doesn't simply remove parts of it

**Comment by cedricdelpoux** [comment]
It strange that `demoteHeadings` option change the behaviour because it should just change the title level. 
For the "bold" case, there is a specific case. Bold is disabled for headings [here](https://github.com/cedricdelpoux/gatsby-source-google-docs/blob/master/utils/google-document.js#L344) to avoid DOM to be `<h2><b>Heading</b></h2>` all the time. I think heading should be style using css `h2 { font-weight: bold }` for example

**Comment by cedricdelpoux** [comment]
Hi @kylekarpack, the issue is resolved in `v2.0.0` version.

With default behaviour, "Part Bold" will no longer disappear.  

But if you really want it to be bold, there is an extra work to do.
You need to remove bold formatting from default Google Docs Headings and save the default formatting, then apply the bold to your headings.

**Comment by kylekarpack** [comment]
Thanks for the update! That will work just fine. Looking forward to trying it out soon.

**Comment by kylekarpack** [comment]
Hi @cedricdelpoux , I tried out the new version and I think all is well, except that the `demoteHeadings: false` option doesn't appear to work any longer. Let me know if a quick repro would be helpful

**Comment by cedricdelpoux** [comment]
Thank you for reporting this! There was a bug with all options. It's now resolved with `2.0.1` version.
